packages/app/src/background/services/credentials/index.ts

```diff
@@ -201,6 +212,105 @@ export default class VerifiableCredentialsService implements IBackupable {
+      options: {
+        title: "Verifiable Presentation generated.",
+        message: `Generated 1 Verifiable Presentation.`,
+        iconUrl: browser.runtime.getURL("/icons/logo.png"),
+        type: "basic",
+      },
+    });
+    const tabs = await browser.tabs.query({ active: true });
+    await Promise.all(
+      tabs.map((tab) =>
+        browser.tabs
+          .sendMessage(tab.id!, {
+            type: EventName.GENERATE_VERIFIABLE_PRESENTATION,
+            payload: { verifiablePresentation },
+          })
+          .catch(() => undefined),
+      ),
+    );
+  };
+
+  generateVerifiablePresentationWithCryptkeeper = async ({
+    verifiablePresentation,
+    address,
+  }: IGenerateVerifiablePresentationWithCryptkeeperArgs): Promise<void> => {
+    const serializedVerifiablePresentation = serializeVerifiablePresentation(verifiablePresentation);
+    const signature = await this.walletService.signMessage({
+      message: serializedVerifiablePresentation,
+      address,
+    });
+    const signedVerifiablePresentation = {
+      ...verifiablePresentation,
+      proof: [
+        {
+          type: [ETHEREUM_SIGNATURE_SPECIFICATION_TYPE],
+          proofPurpose: VERIFIABLE_CREDENTIAL_PROOF_PURPOSE,
+          verificationMethod: address,
+          created: new Date(),
+          proofValue: signature,
+        },
+      ],
+    };
+
+    await this.historyService.trackOperation(OperationType.GENERATE_VERIFIABLE_PRESENTATION, {});
+    await this.notificationService.create({
+      options: {
+        title: "Verifiable Presentation generated.",
+        message: `Generated 1 Verifiable Presentation.`,
+        iconUrl: browser.runtime.getURL("/icons/logo.png"),
+        type: "basic",
+      },
+    });
+    const tabs = await browser.tabs.query({ active: true });
+    await Promise.all(
+      tabs.map((tab) =>
+        browser.tabs
+          .sendMessage(tab.id!, {
+            type: EventName.GENERATE_VERIFIABLE_PRESENTATION,
+            payload: { verifiablePresentation: signedVerifiablePresentation },
+          })
+          .catch(() => undefined),
+      ),
+    );
+  };
+
+  rejectVerifiablePresentationRequest = async (): Promise<void> => {
+    await this.historyService.trackOperation(OperationType.REJECT_VERIFIABLE_PRESENTATION_REQUEST, {});
+    await this.notificationService.create({
+      options: {
+        title: "Request to generate Verifiable Presentation rejected",
+        message: `Rejected a request to generate 1 Verifiable Presentation.`,
+        iconUrl: browser.runtime.getURL("/icons/logo.png"),
+        type: "basic",
+      },
+    });
+    const tabs = await browser.tabs.query({ active: true });
+    await Promise.all(
+      tabs.map((tab) =>
+        browser.tabs
+          .sendMessage(tab.id!, {
+            type: EventName.USER_REJECT,
+            payload: { type: RejectRequests.VERIFIABLE_PRESENTATION_REQUEST },
+          })
+          .catch(() => undefined),
+      ),
+    );
```

Comment:
The same is here for pushEvent.

Reply:
Same as above, will include this in my refactors.